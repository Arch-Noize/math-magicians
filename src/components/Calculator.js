import React, { useState } from 'react';
import './styles/styles.css';
import Button from './Button';
import calculate from '../logic/calculate';

const numbers = ['AC', '+/-', '%', '\u00F7', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
const Calculator = () => {
  const [calc, setCalc] = useState({ total: null, next: null, operation: null });
  const { total, next, operation } = calc;

  const handleEvent = (e) => {
    const target = e.target.textContent;
    setCalc(calculate(calc, target));
  };
  return (
    <div className="calc-container">
      <h2>Lets do some math!</h2>
      <div className="calculator">
        <div className="result">{next || total || operation || 0}</div>
        {numbers.map((number) => <Button number={number} key={number} handleEvent={handleEvent} />)}
      </div>
    </div>
  );
};

export default Calculator;
