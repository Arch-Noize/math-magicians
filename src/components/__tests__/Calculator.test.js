import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Loading calculator component', () => {
  it('renders correctly', () => {
    const { tree } = render(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
});
