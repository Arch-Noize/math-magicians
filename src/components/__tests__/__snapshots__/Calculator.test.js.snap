// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading calculator component renders correctly 1`] = `undefined`;
